test/integration/scheduler/queue_test.go

```diff
@@ -805,28 +805,34 @@ func TestCoreResourceEnqueue(t *testing.T) {
-				st.MakeNode().Name("fake-node2").Label("node", "fake-node").Obj(),
+				st.MakeNode().Name("fake-node1").Label("node", "fake-node").Label("region", "fake-node").Obj(),
+				st.MakeNode().Name("fake-node2").Label("node", "fake-node").Label("region", "fake-node").Obj(),
 			},
 			initialPods: []*v1.Pod{
 				st.MakePod().Name("pod1").Label("key1", "val").SpreadConstraint(1, "node", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), nil, nil, nil, nil).Container("image").Node("fake-node1").Obj(),
 				st.MakePod().Name("pod2").Label("key1", "val").SpreadConstraint(1, "zone", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), nil, nil, nil, nil).Container("image").Node("fake-node2").Obj(),
+				st.MakePod().Name("pod3").Label("key1", "val").SpreadConstraint(1, "region", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), nil, nil, nil, nil).Container("image").Node("fake-node2").Obj(),
 			},
 			pods: []*v1.Pod{
-				// - Pod3 and Pod4 will be rejected by the PodTopologySpread plugin.
-				st.MakePod().Name("pod3").Label("key1", "val").SpreadConstraint(1, "node", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), ptr.To(int32(3)), nil, nil, nil).Container("image").Obj(),
-				st.MakePod().Name("pod4").Label("key1", "val").SpreadConstraint(1, "zone", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), ptr.To(int32(3)), nil, nil, nil).Container("image").Obj(),
+				// - Pod4, Pod5, Pod6, Pod7 will be rejected by the PodTopologySpread plugin.
+				st.MakePod().Name("pod4").Label("key1", "val").SpreadConstraint(1, "node", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), ptr.To(int32(3)), nil, nil, nil).Container("image").Obj(),
+				st.MakePod().Name("pod5").Label("key1", "val").SpreadConstraint(1, "zone", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), ptr.To(int32(3)), nil, nil, nil).Container("image").Obj(),
+				st.MakePod().Name("pod6").Label("key1", "val").SpreadConstraint(1, "region", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), ptr.To(int32(3)), nil, nil, nil).Container("image").Obj(),
+				st.MakePod().Name("pod7").Label("key1", "val").SpreadConstraint(1, "other", v1.DoNotSchedule, st.MakeLabelSelector().Exists("key1").Obj(), ptr.To(int32(3)), nil, nil, nil).Container("image").Obj(),
 			},
 			triggerFn: func(testCtx *testutils.TestContext) (map[framework.ClusterEvent]uint64, error) {
 				// Trigger an Node update event.
-				// It should requeue pod4 only because this node only has zone label, and doesn't have node label that pod3 requires.
-				node := st.MakeNode().Name("fake-node2").Label("zone", "fake-node").Obj()
+				// It should requeue pod4 because it deletes the "node" label from fake-node2.
+				// It should requeue pod5 because the update adds the "zone" label to fake-node2.
+				// It should not requeue pod6 because the update does not change the value of "region" label.
```

Comment:
We need one more pod that has `topologyKey: country` (or whatever), `fake-node2` was in Japan first, and the update moves it to China. So, basically match -> match case similar to pod6, but the label value is updated.

Reply:
done. I add pod9 (match -> match, and update topologyKey's value)